app/validators/expense_validator.rb

```diff
@@ -73,8 +73,8 @@ def validate_calculated_distance
   def validate_mileage_rate_id
     if @record.car_travel? || @record.bike_travel?
       validate_presence(:mileage_rate_id, :blank)
-      unless @record.mileage_rate_id.nil?
-        add_error(:mileage_rate_id, :invalid) if car_travel_missing_milage_rates || bike_travel_missing_milage_rates
+      if !@record.mileage_rate_id.nil? && (car_travel_missing_milage_rates || bike_travel_missing_milage_rates)
```

Comment:
It is possible that `@record.mileage_rate_id.present?` could be used instead of `!@record.mileage_rate_id.nil?`, although I am not sure if it is an exact equivalent.